frontend/packages/dev-console/src/components/monitoring/alerts/MonitoringAlerts.tsx

```diff
@@ -0,0 +1,232 @@
+      });
+    rls.alerts &&
+      _.map(rls.alerts, (alert: Alert) => {
+        rows.push({
+          parent: _.findIndex(rows, (r) => r.cells[0] === rls.name),
+          fullWidth: true,
+          cells: [
+            {
+              title: (
+                <div className="odc-monitoring-alerts--alert-title">{alertDescription(alert)}</div>
+              ),
+              props: { colSpan: 3 },
+            },
+            {
+              title: (
+                <div className="odc-monitoring-alerts--state-column">
+                  <AlertState state={alertState(alert)} />
+                </div>
+              ),
+              props: { colSpan: 2 },
+            },
+          ],
+        });
+      });
+  });
+  return rows;
+};
+
+const MonitoringAlerts: React.FC<MonitoringAlertsProps> = () => {
+  const {
+    'rowFilter-alert-state': alertStateParams,
+    'rowFilter-alert-severity': alertSeverityParams,
+    name,
+  } = getURLSearchParams();
+  const [sortBy, setSortBy] = React.useState({});
+  const [rows, setRows] = React.useState([]);
+  const dispatch = useDispatch();
+  const rule = useSelector((state: RootState) => state.UI.getIn(['monitoring', 'devRules']));
+
+  // fetch and store the rules in store
+  React.useEffect(() => {
+    dispatch(monitoringSetRules('devRules', getAlertsAndRules(rules?.data).rules));
+  }, [dispatch]);
+
+  const filteredRules = React.useMemo(() => {
+    if (alertStateParams || name || alertSeverityParams) {
+      return _.filter(
+        rule,
+        (r) =>
+          (name && _.includes(_.lowerCase(r.name), _.lowerCase(name))) ||
+          _.includes(alertStateParams, _.lowerCase(r?.state)) ||
+          _.includes(alertSeverityParams, r?.labels.severity),
+      );
+    }
+    return rule;
+  }, [alertSeverityParams, alertStateParams, name, rule]);
+  React.useEffect(() => {
+    setRows(monitoringAlertRows(filteredRules));
+  }, [filteredRules]);
+
+  const onCollapse = (event: React.MouseEvent, rowKey: number, isOpen: boolean) => {
+    rows[rowKey].isOpen = isOpen;
+    setRows([...rows]);
+  };
+
+  const alertStateReducer = (alert) => _.toLower(alert?.state);
+  const alertSeverityReducer = (alert) => alert?.labels?.severity;
+
+  const alertStates = [AlertStates.firing, AlertStates.pending, AlertStates.silenced];
+
+  const severities = [Severities.critical, Severities.warning];
+
+  const filters = [
+    {
+      filterGroupName: 'Alert State',
+      type: 'alert-state',
+      reducer: alertStateReducer,
+      items: _.map(alertStates, (state) => ({
+        id: state.toLowerCase(),
+        title: state,
+      })),
+    },
+    {
+      filterGroupName: 'Severity',
+      type: 'alert-severity',
+      reducer: alertSeverityReducer,
+      items: _.map(severities, (severity) => ({
+        id: severity.toLowerCase(),
+        title: severity,
+      })),
+    },
+  ];
+
+  const handleSort = (_event: React.MouseEvent, index: number, direction: SortByDirection) => {
```

Comment:
I think it should be possible to reuse the sorting logic from the admin perspective, which would keep the two perspectives consistent and should be less total code.

The admin perspective is also getting better sort by severity: https://github.com/openshift/console/pull/5853

Reply:
Sure, I will reuse the sorting login once it gets into master and will update my PR.

Reply:
https://github.com/openshift/console/pull/5853 has now merged

Reply:
FYI https://github.com/openshift/console/pull/5527 will switch the admin perspective pages over to the new filter component